### DISCUSSION TITLE
如何在本地用https的url而不是ip访问项目

### DISCUSSION THREAD

**Discussion by GuanJdoJ** [umijs/umi#12070] [Q&A]
1 使用mkcert生成了两个自签证书 `test.local.pem` `test.local-key.pem`
```shell 
$ mkcert test.local
```
2 按照文档配置：https://umijs.org/docs/api/config#https
3 能正常启动，但是只能通过localhost和ip访问https，无法通过`https://test.local`访问项目
![image](https://github.com/umijs/umi/assets/30386473/d09abcb8-f72d-4755-b204-4925aa5d95d9)

请问我该怎么通过`https://test.local`域名访问？

**Comment by fz6m** [comment]
`https` 这个选项是为了本地开发用 https 连接（为了 wasm 操作 array buffer 等情况的安全策略要求），或者为了云 IDE 等远程开发时安全连接才会用到的。

当你没有强诉求迫不得已需要 https 连接，都不应该使用 https 连接进行本地开发，包括世界上最大、使用人数最多的前端框架 nextjs 也是实验性的支持 https 开发而已，所以 99% 的情况你是不需要使用 https 的。

当你真的需要使用 https 连接时，除了生成并配置本地 https 证书，还需要具有正常合理的访问方式，默认情况下可以通过 localhost 的 https 连接本地访问项目，当你需要其他域名在本地访问时，需要一种代理工具将本地对这个虚拟域名的访问代理到 localhost ，以实现一些 cookie 等权限的自动携带、通过域名自动识别逻辑等，从而解决某些问题；关于代理工具，可以使用 charles 、whistle 、switch omega 等，具体请自行学习，有一些代理工具（如 charles ）可以在代理层就提供 https ，请自行研究。
